main/pipeline/pipeline.cc

```diff
@@ -979,6 +982,7 @@ ast::ParsedFilesOrCancelled typecheck(unique_ptr<core::GlobalState> &gs, vector<
                     threadResult.counters = getAndClearThreadCounters();
                     resultq->push(move(threadResult), processedByThread);
                 }
+                workerCounter.DecrementCount();
```

Comment:
This requires all threads to ever pickup a job.
This means that you're asking for worst case performance of scheduler across cores.
Can we have a scheme that doesn't require that?

Reply:
Also note that we _pin_ threads to cores. 
If some other process has higher priority on that core, you may wait arbitrarily long.